pkg/registry/core/service/storage/storage.go

```diff
@@ -636,10 +636,7 @@ func patchAllocatedValues(after After, before Before) {
 }
 
 func needsClusterIP(svc *api.Service) bool {
-	if svc.Spec.Type == api.ServiceTypeExternalName {
-		return false
-	}
-	return true
+	return svc.Spec.Type != api.ServiceTypeExternalName
```

Comment:
not sure we can call that code optimization, imho it just makes the code harder to read